demo/api/src/user-permission/config.service.ts

```diff
@@ -0,0 +1,60 @@
+import { FindUsersArgs, User, UserPermissionConfigInterface, USERPERMISSIONS } from "@comet/cms-api";
+import { Injectable } from "@nestjs/common";
+
+export const staticUsers: User[] = [
+    {
+        id: "1",
+        name: "Admin",
+        email: "demo@comet-dxp.com",
+        language: "en",
+    },
+    {
+        id: "2",
+        name: "Non-Admin",
+        email: "test@test.com",
+        language: "en",
+    },
+];
+
+@Injectable()
+export class UserPermissionConfigService implements UserPermissionConfigInterface {
```

Comment:
For sure there will be a PR in starter, but this is out of scope for this PR.

However, due to the usage of forRootAsync it is possible to use DI in the config.

Reply:
> For sure there will be a PR in starter, but this is out of scope for this PR.

it's hard for me to review this properly if I don't know how it will be used in real project

Reply:
This module provides a useFactory-function like other dynamically configurabale NestJS-modules. For example if you wanted to inject a User-Repository for the config you can do it like this:
```
UserPermissionsModule.forRootAsync({
    useFactory: async (usersService: UsersService) => ({
        config: createUserPermissionConfig(usersService),
    }),
    imports: [UsersModule],
    inject: [UsersService],
}),
```

So nothing special going on here. Do you think this is not sufficient?

Reply:
hm, I was hoping for a pattern where UserPermissionConfig is `@Injectable` and get request it's own dependencies directly and is a service (not a factory like in your example)

But I don't know if/how that is possible

Reply:
What do you think of this? https://github.com/vivid-planet/comet/pull/1251

Reply:
https://github.com/vivid-planet/comet/pull/1223/commits/a5f7f49aa5e8ee50c72ec309c21d873264069f15